### PR TITLE
Memory practice

### PR DESCRIPTION
**Pull request by patarapornkan**

### CODE REVIEW

File: src/memoryPractice/dto/memoryPractice.dto.ts [hunk 1]
```diff
@@ -127,4 +128,138 @@ export class DeleteQuestionDto extends ElderlyWithCaretakerDto {
...
+  @ApiProperty()
+  @IsString()
+  question: string;
+
+  @ApiProperty()
+  @IsString()
+  @IsOptional()
+  imageid?: string;
+
+  @ApiProperty()
```
Comment (ManapatWeeramongkolkul) [inline src/memoryPractice/dto/memoryPractice.dto.ts]: @ApiProperty({ type: multipleChoiceQuestion })

File: src/memoryPractice/dto/memoryPractice.dto.ts [hunk 2]
```diff
@@ -127,4 +128,138 @@ export class DeleteQuestionDto extends ElderlyWithCaretakerDto {
...
+  @IsOptional()
+  multipleChoiceQuestion?: multipleChoiceQuestion;
+
+  @ApiProperty()
+  @IsBoolean()
+  isMultipleChoice: boolean;
+}
+
+export class MemoryPracticeQuestionSetDto{
+  @ApiProperty({ type: [MemoryPracticeQuestion]})
```
Comment (ManapatWeeramongkolkul) [inline src/memoryPractice/dto/memoryPractice.dto.ts:178]: @IsArray()

File: src/memoryPractice/dto/memoryPractice.dto.ts [hunk 3]
```diff
@@ -127,4 +128,138 @@ export class DeleteQuestionDto extends ElderlyWithCaretakerDto {
...
+
+  @ApiProperty()
+  @IsString()
+  answer: string;
+}
+
+export class CreateElderlyAnswersDto{
+  @ApiProperty({ type: [ElderlyAnswer]})
+  @ValidateNested()
+  @Type(() => ElderlyAnswer)
```
Comment (ManapatWeeramongkolkul) [inline src/memoryPractice/dto/memoryPractice.dto.ts:196]: Why is there type needed twice here?

File: src/memoryPractice/dto/memoryPractice.dto.ts [hunk 4]
```diff
@@ -127,4 +128,138 @@ export class DeleteQuestionDto extends ElderlyWithCaretakerDto {
...
+  answers: ElderlyAnswer[]
+}
+
+export class GetHistoryDto{
+  @ApiProperty({ type: [String]})
```
Comment (ManapatWeeramongkolkul) [inline src/memoryPractice/dto/memoryPractice.dto.ts:201]: @IsArray()

File: src/memoryPractice/memoryPractice.controller.ts [hunk 5]
```diff
@@ -91,4 +100,64 @@ export class MemoryPracticeController {
       message: "Question deleted successfully"
     })
   }
+
+  @Get('/question-set/')
```
Comment (ManapatWeeramongkolkul) [inline src/memoryPractice/memoryPractice.controller.ts]: Change the paths to camelcase? Ex: questionSet

File: src/memoryPractice/memoryPractice.service.ts [hunk 6]
```diff
@@ -16,6 +25,9 @@ export class MemoryPracticeService {
     private memoryPracticeQuestionRepository: Repository<MemoryPracticeQuestion>,
     @InjectRepository(MultipleChoiceQuestion)
     private multipleChoiceQuestionRepository: Repository<MultipleChoiceQuestion>,
+    @InjectRepository(MemoryPracticeAnswer)
+        private memoryPracticeAnswerRepository: Repository<MemoryPracticeAnswer>,
+        private userService: UserService
```
Comment (ManapatWeeramongkolkul) [inline src/memoryPractice/memoryPractice.service.ts]: Can u put private userService: UserService on top? (Before any @InjectRepository(..)). Got a bug with this before for some reason

**Review by ManapatWeeramongkolkul — CHANGES_REQUESTED** [review]

File: src/memoryPractice/dto/memoryPractice.dto.ts (hunk 3 above)
Comment (patarapornkan) [reply src/memoryPractice/dto/memoryPractice.dto.ts:196]: the first one is for Swagger. the second one ``@Type`` is for the validation pipe.

**Review by ManapatWeeramongkolkul — APPROVED** [review]

**Pull request merged**